### PR TITLE
Update examples with current Python API, and clean up old files

### PR DESCRIPTION
**Pull request by natke**
Update phi-2 and llama Python examples with latest API, both high and low level.

Remove old files that are no longer used.

Update README.

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -46,26 +46,38 @@ This step requires `cmake` to be installed.
...
+      mv onnxruntime-linux-x64-gpu-1.17.0/include .
+      mv onnxruntime-linux-x64-gpu-1.17.0/lib .
+      ```
+
+    * Or copy equivalent files from a local source build
+      ```
+      git clone https://github.com/microsoft/onnxruntime.git
+      cd onnxruntime
+      ./build.sh --build_shared_lib --skip_tests --parallel [--use_cuda] 
+      ```
```
Comment (baijumeswani) [inline README.md:72]: Do you want to show steps on how and where to copy the libs from and to?

File: examples/llama/llama-loop.py [hunk 2]
```diff
@@ -6,30 +6,34 @@
...
-input_tokens = tokenizer.encode(prompt)
+input_tokens = tokenizer.encode_batch(prompts)
 
-params=og.SearchParams(model)
+params=og.GeneratorParams(model)
 params.max_length = 256
-params.input_ids = input_tokens
+params.set_input_sequences(input_tokens)
+
+print("Generator created")
```
Comment (baijumeswani) [inline examples/llama/llama-loop.py:19]: Is the placement of this print statement incorrect?

File: examples/phi2/README.md [hunk 3]
```diff
@@ -1,13 +1,35 @@
...
-python -m onnxruntime_genai.scripts.export.py -m microsoft/phi-2 -e cpu -p int4 -o model/model.onnx
+pip install numpy
+pip install transformers
+pip install torch
+pip install onnx
+pip install onnxruntime
+```
+
+```bash
+python -m onnxruntime_genai.models.builder -m microsoft/phi-2 -e cpu -p int4 -o model/model.onnx
```
Comment (baijumeswani) [inline examples/phi2/README.md:27]: I could be mistaken but isn't the script expecting a directory as an input for the -o flag?

File: examples/phi2/README.md [hunk 4]
```diff
@@ -1,13 +1,35 @@
...
 
+Install the model builder script dependencies
+
 ```bash
-python -m onnxruntime_genai.scripts.export.py -m microsoft/phi-2 -e cpu -p int4 -o model/model.onnx
+pip install numpy
+pip install transformers
+pip install torch
+pip install onnx
+pip install onnxruntime
```
Comment (edgchen1) [inline examples/phi2/README.md:23]: is this still mandatory if `onnxruntime_genai` includes the `builder` script? having to manually install dependencies doesn't seem ideal.

File: examples/phi2/phi2.py [hunk 5]
```diff
@@ -14,12 +14,12 @@
...
 tokens = tokenizer.encode(prompt)
 
-params=og.SearchParams(model)
+params=og.GeneratorParams(model)
 params.max_length = 200
 params.input_ids = tokens
 
 start_time=time.time()
-output_tokens=model.generate(params)
+output_tokens=model.generate(params)[0]
```
Comment (edgchen1) [inline examples/phi2/phi2.py:22]: does a similar change need to be made in llama.py?

File: examples/phi2/README.md (hunk 3 above)
Comment (natke) [reply examples/phi2/README.md]: I hadn't pushed commit to update this

File: examples/llama/llama-loop.py (hunk 2 above)
Comment (natke) [reply examples/llama/llama-loop.py]: Fixed

File: examples/phi2/README.md (hunk 3 above)
Comment (natke) [reply examples/phi2/README.md]: Fixed

File: examples/phi2/README.md (hunk 4 above)
Comment (edgchen1) [reply examples/phi2/README.md:23]: say I run `pip install onnxruntime_genai`. then I'd expect to be able to use it (including the builder) without installing additional dependencies.

I'm not so experienced with python packaging, but FWIW I think onnxruntime uses `setup(install_requires=...)` to do this.
https://github.com/microsoft/onnxruntime/blob/efbe2b84556c195e7d7f3353321eb3f410a1e645/setup.py#L695

File: examples/llama/README.md [hunk 6]
```diff
@@ -1,9 +1,41 @@
...
+
+```bash
+pip install numpy
+pip install transformers
+pip install torch
+pip install onnx
+pip install onnxruntime
+```
+
+Run the model builder script to load, optimize, quantize and export the model. More details can be found [here](../../src/python/py/models/README.md)
```
Comment (kunal-vaishnavi) [inline examples/llama/README.md:34]:
```suggestion
Run the model builder script to export, optimize, and quantize the model. More details can be found [here](../../src/python/py/models/README.md)
```

File: examples/llama/README.md [hunk 7]
```diff
@@ -1,9 +1,41 @@
...
+
+```bash
+cd examples/phi2
+python -m onnxruntime_genai.models.builder.py -m meta-llama/Llama-2-7b-chat-hf -e cpu -p int4 -o ./model
```
Comment (kunal-vaishnavi) [inline examples/llama/README.md:38]:
```suggestion
python -m onnxruntime_genai.models.builder -m meta-llama/Llama-2-7b-chat-hf -e cpu -p int4 -o ./llama2-7b-chat
```

File: examples/phi2/README.md [hunk 8]
```diff
@@ -1,13 +1,38 @@
...
+```
+
+Run the model builder script to load, optimize, quantize and export the model. More details can be found [here](../../src/python/py/models/README.md)
```
Comment (kunal-vaishnavi) [inline examples/phi2/README.md:26]:
```suggestion
Run the model builder script to export, optimize, and quantize the model. More details can be found [here](../../src/python/py/models/README.md)
```

File: examples/phi2/README.md [hunk 9]
```diff
@@ -1,13 +1,38 @@
...
+
+```bash
+cd examples/llama
```
Comment (kunal-vaishnavi) [inline examples/phi2/README.md:29]:
```suggestion
cd examples/phi2
```

File: examples/llama/README.md (hunk 7 above)
Comment (kunal-vaishnavi) [inline examples/llama/README.md:38]:
```suggestion
python -m onnxruntime_genai.models.builder -m meta-llama/Llama-2-7b-chat-hf -e cpu -p int4 -o ./llama2-7b-chat-int4-cpu
```

File: examples/phi2/README.md [hunk 10]
```diff
@@ -1,13 +1,38 @@
...
+python -m onnxruntime_genai.models.builder -m microsoft/phi-2 -e cpu -p int4 -o ./model
```
Comment (kunal-vaishnavi) [inline examples/phi2/README.md:30]:
```suggestion
python -m onnxruntime_genai.models.builder -m microsoft/phi-2 -e cpu -p int4 -o ./phi2-int4-cpu
```

File: examples/llama/README.md [hunk 11]
```diff
@@ -1,9 +1,41 @@
...
+pip install transformers
+pip install torch
+pip install onnx
+pip install onnxruntime
+```
+
+Run the model builder script to load, optimize, quantize and export the model. More details can be found [here](../../src/python/py/models/README.md)
+
+```bash
+cd examples/phi2
```
Comment (kunal-vaishnavi) [inline examples/llama/README.md:37]:
```suggestion
cd examples/llama
```

File: examples/llama/README.md [hunk 12]
```diff
@@ -1,9 +1,41 @@
 # Llama
 
-## Get model
+## Clone this repo
```
Comment (yufenglee) [inline examples/llama/README.md:3]: We can give user another option to pip install from pypi once we have python package

File: examples/llama/README.md (hunk 7 above)
Comment (yufenglee) [reply examples/llama/README.md:38]: i think you can also remove the model folders as the builder creates config files

File: README.md (hunk 1 above)
Comment (natke) [reply README.md:132]: Added

File: examples/phi2/README.md (hunk 4 above)
Comment (natke) [reply examples/phi2/README.md:23]: We're going to leave it as is for now, as those dependencies are not required for the genai runtime. We will try and think of a better way to do this though.

File: examples/phi2/phi2.py (hunk 5 above)
Comment (natke) [reply examples/phi2/phi2.py]: I made the samples consistent

File: examples/llama/README.md (hunk 12 above)
Comment (natke) [reply examples/llama/README.md:3]: Oh yeah for sure. I wanted to make sure it is correct for what we have currently.

File: examples/phi2/README.md (hunk 10 above)
Comment (kunal-vaishnavi) [reply examples/phi2/README.md:30]: If users try to run multiple examples, then the shared `model` folder may contain files from multiple examples and that may cause unforeseen issues during runtime. Naming the output folder with a specific name in each example would put the files in separate folders and help prevent this.

Comment (natke) [reply examples/phi2/README.md:30]: That's fair. I'll update it

File: examples/llama/README.md (hunk 7 above)
Comment (natke) [reply examples/llama/README.md]: Removed

File: examples/phi2/README.md (hunk 10 above)
Comment (natke) [reply examples/phi2/README.md]: Updated

File: examples/llama/README.md [hunk 13]
```diff
@@ -1,9 +1,41 @@
...
+pip install torch
+pip install onnx
+pip install onnxruntime
+```
+
+Run the model builder script to export, optimize, and quantize the model. More details can be found [here](../../src/python/py/models/README.md)
+
+```bash
+cd examples/phi2
+python -m python -m onnxruntime_genai.models.builder -m meta-llama/Llama-2-7b-chat-hf -e cpu -p int4 -o ./example-models/llama2-7b-chat-int4-cpu
```
Comment (kunal-vaishnavi) [inline examples/llama/README.md:38]:
```suggestion
python -m onnxruntime_genai.models.builder -m meta-llama/Llama-2-7b-chat-hf -e cpu -p int4 -o ./example-models/llama2-7b-chat-int4-cpu
```

**Review by yufenglee — APPROVED** [review]

**Pull request merged**